### ISSUE TITLE
[Help Wanted] make Dragonfly managed by systemd

### ISSUE THREAD

**Issue by Starnop** [dragonflyoss/Dragonfly#1123]
## Why you need it?

Users have to write their own scripts to handle the startup and management of Dragonfly components when they use Dragonfly on physical machines. And systemd provides a complete solution that.

So we can provide **a systemd file**. And later, we can even make our components installed by package management tools such as yum, apt, etc directly.

**Comment by inoc603** [comment]
I can take this. I implemented the deb and rpm packaging script. It would be rather easy to add 2 services. One for dfdamon, and one for supernode. `dfdemon` is fine, but I think `supernode` is not a very good name for a service. The name doesn't show its relationship to the dragonfly project. Maybe we need a better name for it.

**Comment by Starnop** [comment]
> but I think supernode is not a very good name for a service. The name doesn't show its relationship to the dragonfly project.

How about `dfsupernode`? 😢  cc/ @lowzj

**Comment by lowzj** [comment]
agree. I could be named `dfsupernode` in systemd as an abbreviation for dragonfly-supernode.

**Comment by Starnop** [comment]
@inoc603 Any updates?

**Comment by inoc603** [comment]
> @inoc603 Any updates?

I finished the unit files and script. I'll add some test and create a PR this weekend.

**Comment by inoc603** [comment]
@Starnop I ran into some problems testing the service, namely:
- we need default config files to put in the package, probably generated as [issues#1069](https://github.com/dragonflyoss/Dragonfly/issues/1069) suggested.
- in the default config file, we probably need a new default path for the local repo `.small_dragonfly`, for example somewhere under `/opt/dragonfly` other than `$HOME`, since the user may be root when running on the host machine directly.
- supernode requires nginx as dependency, we need to make sure that the installation doesn't conflict with or break existing nginx config on the host.

I'm not quite familiar with the supernode part and I think we should create a branch in the main repo and work on it together. I'll add what I've wrote and how to test the service in a docker container.

**Comment by poblahblahblah** [comment]
Here's what we are using to manage the supernode process in systemd...


/etc/systemd/system/dragonfly-supernode.service:
```
[Unit]
Description = Dragonfly Supernode
Documentation = https://github.com/dragonflyoss/Dragonfly
After = network-online.target
Wants = network-online.target

[Service]
Type = simple
User = nobody
Group = nogroup
Restart = on-failure
LimitNOFILE = 49152
ExecStart = /usr/local/bin/supernode --config /opt/dragonfly/supernode/supernode.yml

[Install]
WantedBy = multi-user.target
```
